src/ContinuousDynamicalSystem/src/FixedBaseDynamics.cpp

```diff
@@ -0,0 +1,182 @@
+        return false;
+    }
+
+    m_kinDyn = kinDyn;
+    m_actuatedDoFs = m_kinDyn->model().getNrOfDOFs();
+
+    // resize matrices
+    m_massMatrix.resize(m_actuatedDoFs + spatialVelocitySize, m_actuatedDoFs + spatialVelocitySize);
+    m_knownCoefficent.resize(m_actuatedDoFs + spatialVelocitySize);
+
+    return true;
+}
+
+bool FixedBaseDynamics::setMassMatrixRegularization(const Eigen::Ref<const Eigen::MatrixXd>& matrix)
+{
+    constexpr auto logPrefix = "[FixedBaseDynamics::setMassMatrixRegularization]";
+
+    if (m_kinDyn == nullptr)
+    {
+        log()->error("{} Please call setKinDyn() before.", logPrefix);
+        return false;
+    }
+
+    if ((m_actuatedDoFs != matrix.rows()) || (matrix.cols() != matrix.rows()))
+    {
+        log()->error("{} The size of the regularization matrix is not correct. The correct size "
+                     "is:  {} x {}. While the input of the function is a {} x {} matrix.",
+                     logPrefix,
+                     m_actuatedDoFs,
+                     m_actuatedDoFs,
+                     matrix.rows(),
+                     matrix.cols());
+        return false;
+    }
+
+    m_massMatrixReglarizationTerm = matrix;
+    m_useMassMatrixRegularizationTerm = true;
+
+    return true;
+}
+
+bool FixedBaseDynamics::dynamics(const double& time, StateDerivative& stateDerivative)
+{
+    constexpr auto logPrefix = "[FixedBaseDynamics::dynamics]";
+
+    if (m_kinDyn == nullptr)
+    {
+        log()->error("{} Please call setKinDyn() before.", logPrefix);
+        return false;
+    }
+
+    // get the state
+    const auto& [jointVelocity, jointPositions] = m_state;
+
+    auto& [jointAcceleration, jointVelocityOutput] = stateDerivative;
+
+    const auto& [jointTorques] = m_controlInput;
+
+    // check the size of the vectors
+    if (jointVelocity.size() != m_actuatedDoFs || jointPositions.size() != m_actuatedDoFs
+        || jointTorques.size() != m_actuatedDoFs)
+    {
+        log()->error("{} Wrong size of the vectors.", logPrefix);
+        return false;
+    }
+
+
+    jointVelocityOutput = jointVelocity;
+
+    // update kindyncomputations object
+    if (!m_kinDyn->setRobotState(jointPositions, jointVelocity, m_gravity))
+    {
+        log()->error("{} Unable to update the kinDyn object.", logPrefix);
+        return false;
+    }
+
+    // compute the mass matrix
+    if (!m_kinDyn->getFreeFloatingMassMatrix(m_massMatrix))
+    {
+        log()->error("{} Unable to get the mass matrix.", logPrefix);
+        return false;
+    }
+
+    // compute the generalized bias forces
+    // here we want to compute the robot acceleration as
+    // robotAcceleration = M^-1 (-h + J' F + tau) = M^-1 * m_knownCoefficent
+
+    if (!m_kinDyn->generalizedBiasForces(m_knownCoefficent))
+    {
+        log()->error("{} Unable to get the bias forces.", logPrefix);
+        return false;
+    }
+
+    m_knownCoefficent *= -1;
+    m_knownCoefficent.tail(m_actuatedDoFs) += jointTorques;
```

Comment:
add a comment here maybe, saying that jointTorques account for both internal and external stress?

Reply:
I think I didn't understand your comment. Here actually we are interested only in the joint dynamics in case of zero external wrench applied on the system

Reply:
what I meant to say was to add a comment saying that J'f = 0 in this case.

Reply:
> what I meant to say was to add a comment saying that J'f = 0 in this case, but clearly I wrote something else. ;D